code/modules/modular_computers/file_system/programs/command/teleporter.dm

```diff
@@ -0,0 +1,118 @@
+/datum/computer_file/program/teleporter
+	filename = "teleporter"
+	filedesc = "Teleporter Control"
+	extended_desc = "A NanoTrasen command remote teleportation hub controller."
+	program_icon_state = "comm"
+	color = LIGHT_COLOR_BLUE
+	size = 8
+	requires_ntnet = TRUE
+	available_on_ntnet = FALSE
+	required_access_run = access_heads
+	usage_flags = PROGRAM_LAPTOP
+	var/obj/machinery/computer/teleporter/linked_comp
+
+/datum/computer_file/program/teleporter/ui_interact(var/mob/user)
+	var/datum/vueui/ui = SSvueui.get_open_ui(user, src)
+	if(!ui)
+		ui = new /datum/vueui/modularcomputer(user, src, "mcomputer-command-teleporter", 600, 400, "Teleporter Control System")
+		ui.auto_update_content = TRUE
+	ui.open()
+
+/datum/computer_file/program/teleporter/vueui_transfer(oldobj)
+	. = FALSE
+	var/uis = SSvueui.transfer_uis(oldobj, src, "mcomputer-command-teleporter", 600, 400, "Teleporter Control System")
+	for(var/tui in uis)
+		var/datum/vueui/ui = tui
+		ui.auto_update_content = TRUE
+		. = TRUE
+
+/datum/computer_file/program/teleporter/vueui_data_change(var/list/data, var/mob/user, var/datum/vueui/ui)
+	. = ..()
+	data = . || data || list()
+
+	// Gather data for computer header
+	var/headerdata = get_header_data(data["_PC"])
+	if(headerdata)
+		data["_PC"] = headerdata
+		. = data
+
+	var/turf/T = get_turf(computer.loc)
+	if(QDELETED(linked_comp))
+		linked_comp = null
+	data["has_linked_comp"] = !!linked_comp
+	if(!linked_comp)
+		var/list/near_comps_info = list()
+		for(var/obj/machinery/computer/teleporter/CT in range(3, T))
+			var/list/comp_info = list(
+				"comp_name" = CT.name,
+				"ref" = "\ref[CT]"
+				)
+			near_comps_info[++near_comps_info.len] = comp_info
+		data["nearby_comps"] = near_comps_info
+	else
+		data["comp_locked_in"] = !!linked_comp.locked
+		var/list/area_index = list()
+
+		var/list/teleport_beacon_info = list()
+		for(var/obj/item/device/radio/beacon/R in teleportbeacons)
+			var/turf/BT = get_turf(R)
+			if(!BT)
+				continue
+			if(isNotStationLevel(BT.z))
+				continue
+			var/tmpname = BT.loc.name
+			if(area_index[tmpname])
+				tmpname = "[tmpname] ([++area_index[tmpname]])"
+			else
+				area_index[tmpname] = 1
+			var/list/teleporter_info = list(
+				"beacon_name" = tmpname,
+				"ref" = "\ref[R]"
+				)
+			teleport_beacon_info[++teleport_beacon_info.len] = teleporter_info
+		data["teleport_beacons"] = teleport_beacon_info
```

Comment:
So vueui migth have issue with reactivity here. 
This if statement conditionally sends data what is no no for root level data
```suggestion
	data["nearby_comps"] = list()
	data["teleport_beacons"] = list()
	if(!linked_comp)
		for(var/obj/machinery/computer/teleporter/CT in range(3, T))
			var/list/comp_info = list(s
				"comp_name" = CT.name,
				"ref" = "\ref[CT]"
				)
			data["nearby_comps"][++data["nearby_comps"].len] = comp_info
	else
		data["comp_locked_in"] = !!linked_comp.locked
		var/list/area_index = list()

		var/list/teleport_beacon_info = list()
		for(var/obj/item/device/radio/beacon/R in teleportbeacons)
			var/turf/BT = get_turf(R)
			if(!BT)
				continue
			if(isNotStationLevel(BT.z))
				continue
			var/tmpname = BT.loc.name
			if(area_index[tmpname])
				tmpname = "[tmpname] ([++area_index[tmpname]])"
			else
				area_index[tmpname] = 1
			var/list/teleporter_info = list(
				"beacon_name" = tmpname,
				"ref" = "\ref[R]"
				)
			teleport_beacon_info[++teleport_beacon_info.len] = teleporter_info
		data["teleport_beacons"] = teleport_beacon_info
```